Assets/MRTK/Core/Inspectors/Profiles/MixedRealityPointerProfileInspector.cs

```diff
@@ -127,11 +126,13 @@ private void RenderPointerList(SerializedProperty list)
                 var controllerType = newPointerOption.FindPropertyRelative("controllerType");
                 var handedness = newPointerOption.FindPropertyRelative("handedness");
                 var prefab = newPointerOption.FindPropertyRelative("pointerPrefab");
+                var raycastLayerMask = newPointerOption.FindPropertyRelative("prioritizedLayerMasks");
```

Comment:
Does this need to be serialized into the `PointerOption`? Can we just render the serializedField directly out of the prefab?

Reply:
Not all prefabs have a serialized backing field, we need to store it inside the profile itself and keep the prefab in lock-step with the profile if the user configures it.

Reply:
Anywhere we need to sync the same piece of information from one place to another makes me uneasy. I've been meaning to investigate drawing an inspector field for the property (regardless of if it has a serialized backing field, which this approach would do too, i think?). That's probably my main concern

Reply:
100% agree with @keveleigh. Duplicate data typically devolves into conflicting values and a need to decide the 'correct' set, either by toolkit logic or developer intervention.

Reply:
I talked to Kurtis a bit offline today, and it's a cursed problem. For pointers that didn't specify an override (ie. shellhandray) there was an implicit reliance on the Focus Provider's default raycast layermask values. Because there's no functional way for us to identify pointers which have an implementation vs. those without, we have to choose some uniform way to allow the user to configure (and save) the raycast layer masks, which is ultimately as a serialized field.

Though the duplicated data isn't the best approach, the approach we've done so far of allowing pointers to self-govern themselves has been met with a lot of confusion and the root cause of several of the issues we've had filed. For minute to minute interactions the pointer's own specification for their prioritized layer mask will take precedence. The purpose of the new serialized field in the editor is to explicitly reveal and provide a clear way for users to configure those pointers which had their PrioritizedLayerMask implementation undefined. For those that do specify their own definitions, the profile will adopt and properly reflect the changes back onto the prefab. 

Though drawing inspectors for properties irregardless of a backing field would reduce the redundancy, it introduces a new problem of changes potentially not getting saved for those instances where there is nothing serialized in the backend, which while "technically" not a regression in behavior, is much more confusing and unfriendly experience. The input system in v2 is already the "grand administrator" for input related activities, so I think our more advanced users have the expectation that the input profile should be a "one-stop-shop" for adjusting high-level input settings, opting to avoid looking at the pointer if possible.